Managers/LuaMan.cpp

```diff
@@ -1,2829 +1,474 @@
-}
+	void LuaMan::SavePointerAsGlobal(void *objectToSave, const std::string &globalName) {
+		// Push the pointer onto the Lua stack.
+		lua_pushlightuserdata(m_MasterState, objectToSave);
+		// Pop and assign that pointer to a global var in the Lua state.
+		lua_setglobal(m_MasterState, globalName.c_str());
+	}
 
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-void LuaMan::Update()
-{
-	lua_gc(m_pMasterState, LUA_GCSTEP, 1);
-}
+	bool LuaMan::GlobalIsDefined(const std::string &globalName) {
+		// Get the var you want onto the stack so we can check it.
+		lua_getglobal(m_MasterState, globalName.c_str());
+		// Now report if it is nil/null or not.
+		bool isDefined = !lua_isnil(m_MasterState, -1);
+		// Pop the var so this operation is balanced and leaves the stack as it was.
+		lua_pop(m_MasterState, 1);
 
-/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+		return isDefined;
+	}
 
-int LuaMan::FileOpen(std::string filename, std::string mode)
-{
-	int fl = -1;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	// Find some suitable file
-	for (int i = 0; i < MAX_OPEN_FILES; ++i)
-		if (m_Files[i] == 0)
-		{
-			fl = i;
-			break;
+	bool LuaMan::TableEntryIsDefined(const std::string &tableName, const std::string &indexName) {
+		// Push the table onto the stack, checking if it even exists.
+		lua_getglobal(m_MasterState, tableName.c_str());
+		if (!lua_istable(m_MasterState, -1)) {
+			// Clean up and report that there was nothing properly defined here.
+			lua_pop(m_MasterState, 1);
+			return false;
 		}
-
-	if (fl == -1)
-	{
-		g_ConsoleMan.PrintString("Error: Can't open file, no more slots.");
-		return -1;
+		// Push the value at the requested index onto the stack so we can check if it's anything.
+		lua_getfield(m_MasterState, -1, indexName.c_str());
+		// Now report if it is nil/null or not
+		bool isDefined = !lua_isnil(m_MasterState, -1);
+		// Pop both the var and the table so this operation is balanced and leaves the stack as it was.
+		lua_pop(m_MasterState, 2);
+
+		return isDefined;
 	}
 
-	//Check for path back-traversing and .rte extension. Everything is allowed to read or write only inside rte's
-    string dotString = "..";
-	string rteString = ".rte";
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
-	string fullPath = System::GetWorkingDirectory() + filename;
+	int LuaMan::FileOpen(const std::string &fileName, const std::string &accessMode) {
+		int fileIndex = -1;
+		for (int i = 0; i < c_MaxOpenFiles; ++i) {
+			if (!m_OpenedFiles.at(i)) {
+				fileIndex = i;
+				break;
+			}
+		}
+		if (fileIndex == -1) {
+			g_ConsoleMan.PrintString("Error: Can't open file, maximum number of files already open.");
```

Comment:
I wonder if there's any real reason to have a MaxOpenFiles setup. I guess just to avoid memory leaks from people not closing files?
Cause like, we could easily get replace this with just a vector or a list or whatever without much effort, but maybe it's actually a feature not a limitation?

Reply:
Ehh
I don't know and I honestly don't care too much.
I guess it's a limit to prevent opening 9000 files and trying to do some nasty shit.

Reply:
Fair enough. If someone every complains we can worry about it then.